### ISSUE TITLE
[cds-dbm] - failed to load model undefined

### ISSUE THREAD

**Issue by muellerj93** [mikezaschka/cds-dbm#298]
Hi,

I'm using cds-dbm to deploy my cds schema to a PostgreSQL service instance on our BTP account.
This has worked for many deployments up until now.
I recently made a change to my cds schema and tried to deploy it again (without changing any config of cds-dbm). This now fails with the error message in the title.

With the CF Logs for the deployer application I was able to track down the following callstack:

cds-dbm deploy [services] (STDERR, APP/TASK/deploy_to_postgresql)#

```
[Thu Feb 16 10:27:39 UTC 2023] at async exports.handler (/home/vcap/deps/1/node_modules/cds-dbm/dist/cli/deploy.js:38:9)
[Thu Feb 16 10:27:39 UTC 2023] at async PostgresAdapter.deploy (/home/vcap/deps/1/node_modules/cds-dbm/dist/adapter/BaseAdapter.js:121:9)
[Thu Feb 16 10:27:39 UTC 2023] at PostgresAdapter.initCds (/home/vcap/deps/1/node_modules/cds-dbm/dist/adapter/BaseAdapter.js:202:19)
[Thu Feb 16 10:27:39 UTC 2023] Error: [cds-dbm] - failed to load model undefined
```

Below you can find my cds / cds-dbm config (last changed 2 months ago, with the latest successful deployment on 10.02.2023):

```
...
"db": {
            "[production]": {
                "kind": "database",
                "dialect": "plain",
                "impl": "cds-pg",
                "model": ["srv"]
            },
...
"build": {
        "tasks": [
            {
                "for": "node-cf",
                "src": "srv",
                "options": {
                    "model": ["db", "srv", "app"]
                }
            },
            {
                "use": "cds-dbm/dist/build/postgres-cf",
                "for": "postgres-cf",
                "src": "db",
                "options": {
                    "deployCmd": "npx cds-dbm deploy --load-via delta --auto-undeploy"
                }
            }
        ]
    },
    "migrations": {
        "db": {
            "schema": {
                "default": "public",
                "clone": "_cdsdbm_clone",
                "reference": "_cdsdbm_ref"
            },
            "deploy": {
                "tmpFile": "tmp/_autodeploy.json",
                "undeployFile": "db/undeploy.json"
            }
        }
    }
```

Additionally an excerpt of my mta.yaml file:
```
build-parameters:
  before-all:
    - builder: custom
      commands:
        - tsc --skipLibCheck
        - npm install --omit=dev --ignore-scripts
        - cds build --production
        - npx rimraf srv/**/*.js
        - npx rimraf srv/**/*.js.map
        - npx rimraf test/**/*.js
        - npx rimraf test/**/*.js.map
        - npx rimraf gen/db/src/gen/data
        # - npx rimraf gen/db/data/schwarz.sci.monitoring-MetricDefinitions.csv
        - cp ./.cdsrc.json ./gen/srv
...
# --------------------- DB-Deployer MODULE -------------------
  - name: argus-sci-monitoring-db-deployer
  # ------------------------------------------------------------
    type: custom
    path: gen/db
    parameters:
      buildpacks: [https://github.com/cloudfoundry/apt-buildpack#v0.2.2, nodejs_buildpack]
      no-route: true # application doesn't need to be accessed after deployment
      no-start: true # application doesn't need to be startable
      disk-quota: 2GB
      memory: 512MB
      tasks:
        - name: deploy_to_postgresql
          command: chmod 755 deploy.sh && ./deploy.sh
          disk-quota: 2GB
          memory: 512MB
    build-parameters:
      ignore: ["node_modules/"]
    requires:
      - name: argus-sci-monitoring-database
...
```

I'm using `@sap/cds version 5.9.8` and `cds-dbm version 0.0.36`.

After unsuccessfully deploying my changed cds model, I changed it back to the original state (the last successfully deployed version), but even that now fails with the same error message.

As my last change to the cds-dbm config was earlier than my last successful deployment with above config, I'm really confused as to where the error might be hiding.

Can you help me to track down the error?
Do you need further information?

Best regards & thanks in advance

**Comment by ryegrosT8** [comment]
Hi there,

cds-dbm is currently using @sap/cds 6.5.0 now, I think that using version 5.9.8 in the project could lead to some collisions. Did you try to upgrade your @sap/cds and also cds-pg to the most recent version?

Kind regards.

**Comment by muellerj93** [comment]
Hi,
the cds-dbm version I'm using (0.0.36) is using @sap/cds 5.8.3 (according to my package-lock.json), so in theory there shouldn't be any collisions.

However, I'll still try to update to the latest versions and check if the deployment works.

Regards

**Comment by mokgavdilabs** [comment]
Just had the same issue. I changed it to fixed version to keep at stable:
"@sap/cds": "5.9.8",
"cds-dbm": "0.0.36",
"cds-pg": "0.1.33",

**Comment by mikezaschka** [comment]
Fixed with #300, merged and released in version 1.11.

Please update to `@sap/cap` 6.5 and the latest versions of `cds-pg` and `cds-dbm`.
